### PR TITLE
Use rector downgrade php

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
It seems got error:

```bash
0s
Run bin/rector process rules --ansi
 [ERROR] File path                                                              
         "/home/runner/work/rector-src/rector-src/vendor/rector/rector-downgrade
         -php/config/config.php" was not found while creating                   
         "Symplify\SmartFileSystem\SmartFileInfo" object.     
```

**Comment by TomasVotruba** [comment]
Possibly this is missing in `composer.json` to make config extension work:
https://github.com/rectorphp/rector-symfony/blob/c6abe8d19e2ebde7365e94eb8bb268b87935bd9d/composer.json#L61-L66

Should be added to rector-downgrade-php `composer.json`

An type as well: https://github.com/rectorphp/rector-symfony/blob/c6abe8d19e2ebde7365e94eb8bb268b87935bd9d/composer.json#L3

**Comment by samsonasik** [comment]
https://github.com/rectorphp/rector-downgrade-php/pull/7

**Comment by samsonasik** [comment]
Let's try

File: .github/workflows/packages_tests.yaml [hunk 1]
```diff
@@ -50,6 +51,10 @@ jobs:
                 run: composer require rector/rector-src dev-main#${{github.event.pull_request.head.sha}} --no-update
                 if: ${{ github.event_name == 'pull_request' }}
 
+            -
+                run: composer require rector/rector-downgrade-php:dev-main --no-update
+                if: github.repository != 'rectorphp/rector-downgrade-php'
```
Comment (samsonasik) [inline .github/workflows/packages_tests.yaml]: this is temporary as require rector/rector-src in previous step seems using --no-update

**Comment by samsonasik** [comment]
@TomasVotruba I am not sure about error on packages-tests workflow for `rectorphp/rector-downgrade-php`:

```bash

There were 671 errors:
1) Rector\Tests\DowngradePhp53\Rector\Dir\DirConstToFileConstRector\DirConstToFileConstRectorTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
Symfony\Component\DependencyInjection\Exception\RuntimeException: Cannot autowire service "Rector\DeadCode\Comparator\Parameter\ParameterDefaultsComparator": argument "$defaultParameterValueResolver" of method "__construct()" references class "Rector\DowngradePhp80\Reflection\DefaultParameterValueResolver" but no such service exists.
```

it seems overlapped with previous command for `composer require rector/rector-src dev-main#${{github.event.pull_request.head.sha}} --no-update`

https://github.com/rectorphp/rector-src/runs/7091105254?check_suite_focus=true#step:8:27

**Comment by samsonasik** [comment]
tweaked with check to not require rector/rector-src temporary on package tests for rectorphp/rector-downgrade-php ,it seems due to overlapped service https://github.com/rectorphp/rector-src/pull/2582/commits/132308091a434c1af7481846755dc14d8a88bcef

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Let's roll 🎊💪

File: easy-ci.php [hunk 2]
```diff
@@ -77,8 +85,16 @@
...
         // deprecated, keep it for now
         TemplateResolverInterface::class,
+
+        MethodTypeAnalyzer::class,
+        DoctrineAnnotationFactory::class,
+        ClassFromEnumFactory::class,
+        CoalesceAnalyzer::class,
+        NamespacedNameDecorator::class,
+        NamedVariableFactory::class,
+        BinaryOpTreeRootLocator::class,
```
Comment (TomasVotruba) [inline easy-ci.php:98]: Are any of these used only in the downgrade set? 
If so, those should be moved there, even for price of namespace rename

Comment (samsonasik) [reply easy-ci.php:98]: it seems yes

**Comment by TomasVotruba** [comment]
For ref, this PR makes use of this package: https://github.com/rectorphp/rector-downgrade-php

**Comment by TomasVotruba** [comment]
Thanks to this decoupling, we got ~21 % speed improvement on CI run for rules-test :tada:  :zap: 

<br>

## Before - 3:43

image

<br>

## After - 3:04

image

**Pull request merged**